accounts/views.py

```diff
@@ -0,0 +1,142 @@
+from rest_framework import status
+from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .serializers import LoginSerializer
+from .models import MyUser
+from .serializers import RegistrationSerializer
+from .emailer import send_email
+
+
+class RegistrationAPIView(APIView):
+    """
+    Registers a new user.
+    """
+    serializer_class = RegistrationSerializer
+
+    def post(self, request):
+        """
+        Creates a new User object.
+        Username, email, and password are required.
+        Returns a JSON web token.
+        """
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        email = serializer.validated_data['email']
+        password = serializer.validated_data['password']
+
+        existing_user = MyUser.objects.filter(email__exact=email).first()
+        if existing_user:
+            return Response({'error': 'User with this email is already registered'}, status=status.HTTP_400_BAD_REQUEST)
+
+        user = MyUser.objects.create_user(email)
+        user.set_password(password)
+        token = Token.objects.create(user=user).key
+        user.save()
+
+        try:
+            send_email(email, password)
+        except:
+            pass
+
+        return Response({'email': email, 'id': user.id, 'auth_token': token}, status=status.HTTP_201_CREATED)
+
+
+class LoginAPIView(APIView):
+    serializer_class = LoginSerializer
+
+    def post(self, request):
+        """
+        Checks is user exists.
+        Email and password are required.
+        Returns a JSON web token.
+        """
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        email = serializer.validated_data['email']
+        token = serializer.validated_data['auth_token'].key
+
+        user = MyUser.objects.filter(email__exact=email).first()
```

Comment:
```suggestion
        user = MyUser.objects.filter(email__iexact=email).first()
```
It's always better to use `iexact` filtering for email, verification codes etc, if they are not case sensitive ofc, because usually users are not so smart :grin: - so let's make sure that `jonsnow@westeros.com` and `JonSnow@WESTEROS.com` is the same email address

Reply:
Исправил